.github/workflows/samples-tools/pkg/utils/config.go

```diff
@@ -0,0 +1,101 @@
+package utils
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+type Config struct {
+	// Pattern to match filenames or directories.
+	Match []string `json:"match"`
+
+	// Pattern to ignore filenames or directories.
+	Ignore []string `json:"ignore"`
+
+	// Filename to look for the root of a package.
+	Package []string `json:"package"`
+
+	// Packages to always exclude.
+	ExcludePackages []string `json:"exclude-packages"`
+
+	// Actions to run including the given commands.
+	Actions map[string][]struct {
+		Command string   `json:"command"`
+		Args    []string `json:"args"`
+	} `json:"actions"`
+}
+
+func LoadConfig(path string) (Config, error) {
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return Config{}, err
+	}
+	config, err := ParseConfig(bytes)
+	if err != nil {
+		return Config{}, err
+	}
+	return config, nil
+}
+
+func ParseConfig(source []byte) (Config, error) {
+	var config Config
+	err := json.Unmarshal(StripComments(source), &config)
```

Comment:
future: This will create a broad tooling compatibility problem for the files on disk, such as rendering in IDEs and validating via general JSON parsing.

suggestion: Adopt a format that explicitly supports comments.

Reply:
I decided to change the file extension to `.jsonc`, which is a superset of JSON that allows comments. That way it renders correctly and behaves as expected. A minor caveat is that we only currently support single line comments, and JSONC should also support multi-line comments.